### PR TITLE
Development branch

### PR DESCRIPTION
**Pull request by atoledo-ec**
## Methods feature branch

Every method was created using RUBY

In this project we have created this enumerable methods:

  -  #my_each
  -  #my_each_with_index
  -  #my_select
  -  #my_each
  -  #my_all?
  -  #my_any?
  -  #my_none?
  -  #my_count
  -  #my_map
  -  #my_inject

### CODE REVIEW

**Review by Rhelli — CHANGES_REQUESTED** [review]
# Changes Required 🛑 ✅

Hi guys, well done on the project so far. There are however changes that need to be made to this project in order to advance on and get this project approved.

Before I go over the issues, I would like to address something first - **please do not close Pull Requests between code reviews.** It's extremely helpful for TSEs to be able to see what the previous TSE marking your project has said. Just as the last TSE said, I will echo. Maintain the same Pull Request through multiple code reviews.

Secondly, unfortunately, there are too many errors with this project to go into detail. As currently testing, there are 75 errors 👇🏾

image

Given my 15 minutes allotted time to review your project - I cannot cover all of the errors, however I can give you some advice in fixing them.

### 1. Use a testing script to catch the errors before you re-submit.

[Here is my testing script](https://pastebin.com/WUNBjshC) that I created when I was doing this project. Simply copy and paste your methods in and out and run the script against these tests to verify if your methods are functioning as they should be. The expected solutions from running the tests should be next to the tests themselves but in general, you should see no errors at all.

### 2. Your `my_each` method has to be rock solid.

If your `my_each` does not function correctly for this project, very little else will. Currently your `my_each` has an error in it:

image

Line 13 should read `while i < list.length`, not `while i < element.length`.

### 3. There are numerous undefined arguments

Much like the issue above, there are numerous 'undefined arguments' being used causing errors.

*******

Do not be disheartened by the requested changes. Go back to your script, go over it with a fine comb for obvious errors (i.e. undeclared variables), use the test script I have attached, and resubmit when you're ready. Also, resubmit on this Pull Request, not another one! Good luck.

If you need any help, @ me here on Github or find me on Slack.

All the best,

💂🏽 Rory

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__I must commend your efforts in following the previous reviewer's advice. It really showed that you've worked on his recommendations as this project is near completion. We just have to make some changes and your project should be good to go. Let's get to it.__

**[ REQUIRED CHANGES ]**

- [x] You'll agree with me that the `text.txt` file is not necessary for this project. I understand that you used it in production, but you should have added it to your `.gitignore` file. Kindly remove it.

- [x] Your `my_all?` method failed 2 test cases. See examples below:

**Case 1: When no block or argument is given, `my_all` should return false when one of the collection members is false or nil**
```
false_array = [1, false, 'hi', []]
false_array.all?   # returns rfalse (correct answer)
p false_array.my_all?   # returns true (incorrect)
```

**Case 2: When a class is passed as an argument, `my_all` should return true if all of the collection is a member of such class (in this case, the class is `Integer`)**
```
int_array = [1,4,3,8,6,5]
int_array.all?(Integer)    # returns true (correct answer)
int_array.my_all?(Integer)   #r returns false (incorrect)
```

- [x] Your `my_map` method failed 1 test case. I believe the failure is as a result of an undefined local variable `new_arg` as observed in your code.



Kindly check line 108. Are you trying to use `new_arr` instead of `new_arg`?


**[ COMMENTS ]**
- Kindly make the changes highlighted above.
- You should be careful in making those changes. Do not let the new changes affect the existing flow.

**Please make the required changes and submit another review request when you are done.**

* * *

**_You can reach out to me on Twitter (@ola_lere), Github(@teekaytech), or slack (@taofeekolalere )_**
__Happy Coding :smiley:__

File: methods.rb
Comment (ghost) [inline methods.rb]: `my_none` method should iterate over a range of values.
```
print (0..4).none?  #=> false
print (0..4).my_none?  #=> methods.rb:81:in `my_none?': undefined method `length' for 0..4:Range (NoMethodError)
```

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

### Hello team,

**Great job implementing the requested changes. However, there are a few fixes needed before the project can be approved.**

## Mandatory

1). **Please add text.txt file in .gitignore file.**

2). **There are a few offenses detected by rubocop.**
rubocop

3). **Check one comment below.**

### Happy coding 😃.

**PS: You can reach me on slack (Rose Wanjohi).**

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__I must commend you guys for the job well done. This project is almost there. Almost all the major bugs have been fixed. Right now, just one bug is stopping me from approving this project. Check my comment below.__

**[ REQUIRED CHANGES ]**

- [ ] `my_map` function is failing a test case.

**`my_map` executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.**

Here's the error message:


We can deduce from the error message that a particular undefined method `call` is being applied to an array in your `my_map` function.



This is happening at line 108. What did you intend to achieve with that `call` method? Is it a bug?


**Please make the required changes and submit another review request when you are done.**

* * *

**_You can reach out to me on Twitter (@ola_lere), Github(@teekaytech), or slack (@taofeekolalere )_**
__Happy Coding :smiley:__

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**
